client/index.js

```diff
@@ -65,10 +66,27 @@ async function doRender ({ Component, props, err }) {
     props = await loadGetInitialProps(Component, { err, pathname, query })
   }
 
+  if (Component === ErrorComponent) {
```

Comment:
What happen if we render the Error component twice. Then it might override the actual `lastScroll`
Could you test for that case?